frontend/src/metabase/lib/schema_metadata.js

```diff
@@ -137,9 +137,9 @@ export function getFieldType(field) {
     COORDINATE,
     FOREIGN_KEY,
     PRIMARY_KEY,
-    NUMBER,
     STRING,
     STRING_LIKE,
+    NUMBER,
```

Comment:
This is the real fix, since the earlier types should be the specific ones first (hence, matching `:type/Name` as `STRING`).